src/scripts/main.js

```diff
@@ -1 +1,70 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const timeOut = setTimeout(() => {
+    reject(new Error('First promise was rejected in 3 seconds if not clicked'));
+  }, 3000);
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      clearTimeout(timeOut);
+      resolve('First promise was resolved on a left click in the document');
+    }
+  });
+});
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve('Second promise was resolved');
+    }
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+    resolve('Second promise was resolved');
+  });
+});
+
+let leftClick = false;
+let rightClick = false;
+
+const thirdPromise = new Promise((resolve) => {
+  document.addEventListener('click', (e) => {
+    leftClick = true;
+
+    if (leftClick && rightClick) {
+      resolve('Third promise was resolved');
+
+      leftClick = false;
+      rightClick = false;
+    }
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    rightClick = true;
+
+    if (leftClick && rightClick) {
+      resolve('Third promise was resolved');
+
+      leftClick = false;
+      rightClick = false;
+    }
+  });
+});
```

Comment:
Your third promise logic seems flawed. It will resolve as soon as both left and right clicks have been detected, regardless of the order. However, the task requires the promise to resolve only when a right click follows a left click without any other left clicks in between. You need to adjust your logic to account for this.